tests/Doctrine/Tests/ORM/Functional/Ticket/GH10449Test.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\MappingException;
+use Doctrine\Tests\OrmTestCase;
+
+use function get_parent_class;
+use function method_exists;
+
+class GH10449Test extends OrmTestCase
+{
+    public function testToManyAssociationOnMappedSuperclassShallBeRejected(): void
+    {
+        $em      = $this->getTestEntityManager();
+        $classes = [GH10449MappedSuperclass::class, GH10449Entity::class, GH10449ToManyAssociationTarget::class];
+
+        $this->expectException(MappingException::class);
+        $this->expectExceptionMessageMatches('/illegal to put an inverse side one-to-many or many-to-many association on mapped superclass/');
+
+        foreach ($classes as $class) {
+            $cm = $em->getClassMetadata($class);
+        }
+    }
+
+    /**
+     * Override for BC with PHPUnit <8
+     */
+    public function expectExceptionMessageMatches(string $regularExpression): void
+    {
+        if (method_exists(get_parent_class($this), 'expectExceptionMessageMatches')) {
+            parent::expectExceptionMessageMatches($regularExpression);
+        } else {
+            parent::expectExceptionMessageRegExp($regularExpression);
+        }
+    }
```

Comment:
Would it make sense to move this into a trait? We have a `Doctrine\Tests\PHPUnitCompatibility` namespace for that kind of stuff.

Reply:
Agree, but I'd like to do this as a clean-up in another PR if you don't mind.

TBH I've copied this method from other tests, so we should have a dedicated PR to clean up, not clutter this PR here.